.github/workflows/realm-generator.yml

```diff
@@ -14,32 +14,46 @@ on:
       - .github/workflows/realm-generator.yml
 
 jobs:
-  Generator_CI:
-    runs-on: ${{ matrix.os }}
-
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest] # windows is not tested ue to bug in how error messages are rendered
-        channel: [stable, beta]
-
-    defaults:
-      run:
-        working-directory: ./generator/
+  CI:
+    runs-on: ubuntu-latest
+    env:
+      REALM_CI: true
 
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
 
       - name : Setup Dart SDK
         uses: dart-lang/setup-dart@main
         with:
-          sdk: ${{ matrix.channel }}
+          sdk: stable
 
-      - name: Install dependencies
+      - name: Install generator dependencies
         run: dart pub get
+        working-directory: ./generator/
 
       - name: Run generator tests
-        run: dart test --reporter expanded --coverage ./coverage/
+        run: dart test --reporter expanded --coverage ./coverage/ --test-randomize-ordering-seed random
+        working-directory: ./generator/
```

Comment:
This working directory is set twice. If it is because of the following step I suppose it should be set to realm-dart.

Reply:
this is the working directory of this step. What do you mean it is set twice?